src/backend/distributed/utils/statistics_collection.c

```diff
@@ -145,20 +149,26 @@ CollectBasicUsageStatistics(void)
 		return false;
 	}
 
+#ifndef WIN32
 	unameResult = uname(&unameData);
 	Assert(unameResult == 0);  /* uname() always succeeds if we pass valid buffer */
+#endif
 
 	appendStringInfoString(fields, "{\"citus_version\": ");
 	escape_json(fields, CITUS_VERSION);
 	appendStringInfo(fields, ",\"table_count\": " UINT64_FORMAT, roundedDistTableCount);
 	appendStringInfo(fields, ",\"cluster_size\": " UINT64_FORMAT, roundedClusterSize);
 	appendStringInfo(fields, ",\"worker_node_count\": %u", workerNodeCount);
+#ifndef WIN32
 	appendStringInfoString(fields, ",\"os_name\": ");
 	escape_json(fields, unameData.sysname);
 	appendStringInfoString(fields, ",\"os_release\": ");
 	escape_json(fields, unameData.release);
 	appendStringInfoString(fields, ",\"hwid\": ");
 	escape_json(fields, unameData.machine);
+#else
+	appendStringInfoString(fields, ",\"os_name\": \"Windows\"");
```

Comment:
Take a look at the [Windows System Information Functions](https://msdn.microsoft.com/en-us/library/windows/desktop/ms724953(v=vs.85).aspx). It appears maybe `GetSystemInfo` and `GetProductInfo` can give us what we need pretty easily?

Reply:
Nevermind, I just looked into this and it's not clear what the "right way" is, for whatever reason.

Reply:
I think we actually need `GetVersionEx` to get the numbers to pass to `GetProductInfo`, maybe…

Reply:
OK, so here's how Python does it: https://github.com/python/cpython/blob/aa0735f597b072c0eb00404c4d7df359ddc26755/Python/sysmodule.c#L1050-L1115

This just leaves the architecture. Largely academic, I suppose, but Perl's `POSIX` module has a good implementation (they have an entire `uname` shim for Windows)… https://github.com/Perl/perl5/blob/69374fe705978962b85217f3eb828a93f836fd8d/win32/win32.c#L2109-L2137

Reply:
Some suggested registry keys, which I was hesitant to accept as a solution, but Python seems to fall back to it (https://github.com/python/cpython/blob/fe2d5babba5d26de2093b6518316b268488187be/Lib/platform.py#L500-L545 ). It's always hard to read this stuff because you're never sure what the obsolete paths are and which are modern (Python/Perl tend to have fallback after fallback to determine this stuff, as the APIs seem to have changed both in call style and in semantics over the years).

Reply:
I wrote a replacement for uname inspired by the perl one, thanks for doing this research!